pkg/prometheus/promcfg.go

```diff
@@ -2232,10 +2231,16 @@ func (cg *ConfigGenerator) generateRoleSelectorConfig(k8sSDConfig yaml.MapSlice,
 	labelSelector := labels.SelectorFromValidatedSet(labels.Set(selector.MatchLabels))
 
 	for _, exp := range selector.MatchExpressions {
```

Comment:
Hmm I wonder if we shouldn't use [LabelSelectorAsSelector()](https://pkg.go.dev/k8s.io/apimachinery@v0.32.0/pkg/apis/meta/v1#LabelSelectorAsSelector) to parse the whole selector field instead and call `Requirements()` on the returned value.

Reply:
Actually not because `LabelSelectorAsMap` Don't support LabelSelectorOpNotIn, LabelSelectorOpExists, LabelSelectorOpDoesNotExist

https://github.com/kubernetes/apimachinery/blob/59e9003f02d6f0c8fff53719a7a0604ec82ee9a9/pkg/apis/meta/v1/helpers.go#L96